### DISCUSSION TITLE
Fake Sensor

### DISCUSSION THREAD

**Discussion by shanekuz** [arendst/Tasmota#17870] [Q&A] [answered]
Hi, is any way I can create a fake sensor to send back a Variable calculated in a tasmota rule? I have an Analogue pin connected via 220k resistor to measure battery voltage and have worked out that 810 is fully charges and 510 is where it shuts down so simple (A0-510)/3 is my battery percentage.  I would like this to post in the tele/Tasmota_%%%%%/SENSOR mqtt message showing BatteryPercentage

Any help or ideas on how to do this?

**Comment by barbudor** [comment, marked as answer]
For analog input, did you looked at the Adc Range mode which allow you to remap the ADC 0..1023 range to any range you like ?
I'm using that to remap ADC measure to battery level in mV in one of my sensor

https://tasmota.github.io/docs/ADC/

**Comment by shanekuz** [comment]
Thanks i changed it to ACD Range and the value changed from 795 to 78 i cant find any docs that show how i would set 510 to 0 and 810 to 100 though.   im assuming that's 78% of 1024 ?

**Reply by barbudor** [reply]
Have you read the page I linked ?
Especially the section about the AdcParam command ?

**Comment by Jason2866** [comment]
Exactly this part https://tasmota.github.io/docs/ADC/#commands

**Comment by shanekuz** [comment]
Thank you for your help i think i can get that going from there.
